src/js/other-websites/add_input_icon.js

```diff
@@ -24,9 +24,23 @@ function addRelayMenuToPage(relayMenuWrapper, relayInPageMenu, relayIconBtn) {
 function positionRelayMenu() {
   const relayInPageMenu = document.querySelector(".fx-relay-menu-iframe");
   const relayIconBtn = document.querySelector(".fx-relay-menu-open");
+  const relayInPageMenuIframe = document.querySelector(".fx-relay-menu-iframe");
   const newIconPosition = relayIconBtn.getBoundingClientRect();
-  relayInPageMenu.style.left = newIconPosition.x - 255 + "px";
-  relayInPageMenu.style.top = newIconPosition.top + 40 + "px";
+  const documentPosition = document.documentElement.getBoundingClientRect();
+  
+  // Calculate the "safe area" of add-on in-page menu. If there's not enough room to expand below the icon, it expands above. 
+  const positionMenuBelowIcon = ((((newIconPosition.top - documentPosition.top) - document.documentElement.scrollHeight) * -1) > 400);
```

Comment:
Possibly this could also take into account whether the area above is enough as well? Assuming that the page expands downwards if the menu is too big, that would be better than the situation I just encountered: the menu was displayed on top, and the top part of the menu wasn't visible.

I expect that to be an edge case though, so if that wouldn't make the full menu visible, or if it's too much work, feel free to ignore.

Reply:
I want to leave this as a TODO bc it's an edgecase when there's not enough room above OR below. Which page was this, for reference?

Reply:
imgur.com/register, with my browser devtools open so it would shop up on top.

Reply:
Note: This issue still exists right now. ☹️